sdk/go/common/workspace/paths.go

```diff
@@ -98,8 +98,28 @@ func DetectProjectStackPath(stackName tokens.QName) (string, error) {
 		return "", err
 	}
 
-	return filepath.Join(filepath.Dir(projPath), proj.Config, fmt.Sprintf("%s.%s%s", ProjectFile, qnameFileName(stackName),
-		filepath.Ext(projPath))), nil
+	fileName := fmt.Sprintf("%s.%s%s", ProjectFile, qnameFileName(stackName), filepath.Ext(projPath))
+
+	// Back compat: StacksDirectory used to be called Config.
+	configValue, hasConfigValue := proj.Config.(string)
+	hasConfigValue = hasConfigValue && configValue != ""
+
+	if proj.StacksDirectory != "" {
+		// If config and stacksDirectory are both set return an error
+		if hasConfigValue {
+			return "", fmt.Errorf("can not set `config` and `stacksDirectory`, remove the `config` entry")
+		}
+
+		return filepath.Join(filepath.Dir(projPath), proj.StacksDirectory, fileName), nil
+	}
+
+	// Back compat: If StacksDirectory is not present and Config is given and it's a non-empty string use it
+	// for the stacks directory.
+	if hasConfigValue {
+		return filepath.Join(filepath.Dir(projPath), configValue, fileName), nil
```

Comment:
Should we issue a deprecation warning, suggesting that the use switches to use `StacksDirectory`?